librz/asm/arch/arm/armass64.c

```diff
@@ -1285,222 +1285,118 @@ static bool handlePAC(ut32 *op, const char *str) {
 bool arm64ass(const char *str, ut64 addr, ut32 *op) {
 	ArmOp ops = {0};
 	if (!parseOpcode (str, &ops)) {
+		free (ops.mnemonic);
 		return false;
 	}
 	/* TODO: write tests for this and move out the regsize logic into the mov */
 	if (!strncmp (str, "mov", 3)) {
 		*op = mov (&ops);
-		return *op != -1;
-	}
-	if (!strncmp (str, "cb", 2)) {
+	} else if (!strncmp (str, "cb", 2)) {
```

Comment:
all these strncmp should be replaced with `rz_str_startswith` to be "safer", meaning I don't have to check that the number really corresponds to the length of the string :)

Reply:
Yes,  but it turned out that compiler doesn't optimize strlen for the `const char *` strings:
- Clang 11: https://godbolt.org/z/rnYE38
- GCC 10.2: https://godbolt.org/z/4TGK6n

Reply:
That's probably because our implementation of rz_str_startswith should go in the rz_str.h file and should be static inline. That way the compiler can easily inline it and it will be as good as strncmp with a hard-coded value.

Reply:
Anyway, a change for another day :)